### PR TITLE
yazi: init hm module

### PR DESCRIPTION
**Pull request by xokdvium**
Add initial support for Yazi terminal file manager. The style is based on the upstream catppuccin themes, but with non-color subjective options removed.

Project repo: https://github.com/sxyazi/yazi

### CODE REVIEW

**Comment by xokdvium** [comment]
Here are screenshots with catppuccin-mocha and gruvbox-light. There may some readability issues for modes and help menu for certain themes. But overall seems alright to me. I'd love to hear some input on this theme and how it can be improved

Screenshot from 2024-01-22 17-57-33
Screenshot from 2024-01-22 17-58-07
Screenshot from 2024-01-22 18-06-58

**Comment by xokdvium** [comment]
I'm working on polishing the look. Will upload new screenshots and address all comments.

**Comment by xokdvium** [comment]
Attaching screenshots with the polished module. Contrast for mode selection and percentage is significantly better. Help window also looks much better and is more readable. Theme is `${pkgs.base16-schemes}/share/themes/catppuccin-mocha.yaml`
help
which
folders

**Comment by xokdvium** [comment]
I'm quite happy with how this theme looks, it's quite cohesive with the eza, which uses `ansiterm` colors via stylix alacritty modules. Maybe someone could chime in and test this module with a different setup/terminal emulator/theme.

File: modules/yazi/hm.nix
Comment (xokdvium) [reply modules/yazi/hm.nix:24]: Done

**Review by danth — COMMENTED** [review]
I tested this in GNOME Console, which is not yet themed by Stylix.

This leads to a weird appearance where some theme colors are mixed with non-theme colors from the terminal:

Screenshot of Yazi in GNOME Console

Is there a way we can configure Yazi to, for example, set a background color explicitly rather than relying on the terminal to match the theme? If yes, could we do that? If not then I'll accept merging this as-is for now.

**Comment by xokdvium** [comment]
I've explicitly set all of the available options. Unfortunately there does not seem to be a way to set the default bg color via yazi theme (at least I wasn't able to find a way). Now of the colors match (aside from bg) between stylix-enabled alacritty and gnome console. 

alacritty
gnome

Yazi was recently updated to `v0.2.4`, which contains relevant breaking changes to the theme configuration. Particularly `help.exec` has been renamed to `help.run` with seemingly no back-compat. We should probably wait until the [pr](https://nixpk.gs/pr-tracker.html?pr=294509) makes it to nixos-unstable before merging this. Though this isn't critical and yazi does not seem to mind an invalid option in it's config.

**Comment by danth** [comment]
In that case, I'll go ahead and merge this, since it's working when the terminal is fully themed.

It seems like there's not much we can do to improve on the other case currently. I think the chance of a regular Yazi user relying on the default GNOME console is relatively low, so it's not a *major* issue. However, perhaps we should raise this upstream to get the necessary options added so that we can make our theme more reliable.

**Review by danth — APPROVED** [review]

**Comment by xokdvium** [comment]
Great, feel free to merge to merge this pr in the current state. I've double checked that it works fine on yazi 0.2.4 with the latest changes. I'll investigate the possibility of adding/create an issue upstream regarding the background color.

**Pull request merged**
